lightning/src/chain/keysinterface.rs

```diff
@@ -439,21 +456,92 @@ impl InMemoryChannelKeys {
 	}
 
 	/// Counterparty pubkeys.
-	/// Will panic if on_accept wasn't called.
+	/// Will panic if on_funding_created wasn't called.
 	pub fn counterparty_pubkeys(&self) -> &ChannelPublicKeys { &self.accepted_channel_data.as_ref().unwrap().counterparty_channel_pubkeys }
 
 	/// The contest_delay value specified by our counterparty and applied on holder-broadcastable
 	/// transactions, ie the amount of time that we have to wait to recover our funds if we
 	/// broadcast a transaction. You'll likely want to pass this to the
 	/// ln::chan_utils::build*_transaction functions when signing holder's transactions.
-	/// Will panic if on_accept wasn't called.
+	/// Will panic if on_funding_created wasn't called.
 	pub fn counterparty_selected_contest_delay(&self) -> u16 { self.accepted_channel_data.as_ref().unwrap().counterparty_selected_contest_delay }
 
 	/// The contest_delay value specified by us and applied on transactions broadcastable
 	/// by our counterparty, ie the amount of time that they have to wait to recover their funds
 	/// if they broadcast a transaction.
-	/// Will panic if on_accept wasn't called.
+	/// Will panic if on_funding_created wasn't called.
 	pub fn holder_selected_contest_delay(&self) -> u16 { self.accepted_channel_data.as_ref().unwrap().holder_selected_contest_delay }
+
+	/// Whether the holder is the initiator
+	pub fn is_outbound(&self) -> bool { self.accepted_channel_data.as_ref().unwrap().is_outbound }
+
+	/// Funding outpoint
+	/// Will panic if on_funding_created wasn't called.
+	pub fn funding_outpoint(&self) -> &OutPoint { self.funding_outpoint.as_ref().unwrap() }
+
+	/// Prepare build information for a holder-broadcast commitment tx
+	pub fn build_holder_commitment_info<T: secp256k1::Signing + secp256k1::Verification>(
```

Comment:
What's the purpose of `build_holder_commitment_info`  ? There is a parameters intersection at `build_commitment_tx` as `per_commitment_point` is both required  alone and as part of `CommitmentInfo`.

Reply:
yeah, that code was a bit awkward, changing